cli/src/commands/git/clone.rs

```diff
@@ -178,11 +187,15 @@ fn do_git_clone(
     ui: &mut Ui,
     command: &CommandHelper,
     colocate: bool,
-    depth: Option<NonZeroU32>,
-    remote_name: &str,
-    source: &str,
-    wc_path: &Path,
+    clone_args: CloneArgs<'_>,
 ) -> Result<(WorkspaceCommandHelper, GitFetchStats), CommandError> {
+    let CloneArgs {
+        depth,
+        remote_name,
+        source,
+        wc_path,
+    } = clone_args;
+
     let settings = command.settings_for_new_workspace(wc_path)?;
```

Comment:
nit: instead of packing parameters, I think `do_git_clone()` can be split into two functions: 1. set up workspace (and `workspace_command`), 2. do `git::fetch()`.